cmd/root.go

```diff
@@ -27,63 +29,92 @@ type Scanner interface {
 var level logconfig.Level
 
 // print result of scan as JSON to stdout
-var printJSON bool
+var outputScan bool
 
 // list of namespaces to be skipped from scan
 var skippedNs []string
 
 // skip SSL cert validation when connecting to PolicyServers endpoints
 var insecureSSL bool
 
-var (
-	// rootCmd represents the base command when called without any subcommands
-	rootCmd = &cobra.Command{
-		Use:   "audit-scanner",
-		Short: "Reports evaluation of existing Kubernetes resources with your already deployed Kubewarden policies",
-		Long: `Scans resources in your kubernetes cluster with your already deployed Kubewarden policies.
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "audit-scanner",
+	Short: "Reports evaluation of existing Kubernetes resources with your already deployed Kubewarden policies",
+	Long: `Scans resources in your kubernetes cluster with your already deployed Kubewarden policies.
 Each namespace will have a PolicyReport with the outcome of the scan for resources within this namespace.
 There will be a ClusterPolicyReport with results for cluster-wide resources.`,
 
-		RunE: func(cmd *cobra.Command, args []string) error {
-			level.SetZeroLogLevel()
-			namespace, err := cmd.Flags().GetString("namespace")
-			if err != nil {
-				return err
-			}
-			kubewardenNamespace, err := cmd.Flags().GetString("kubewarden-namespace")
-			if err != nil {
-				return err
-			}
-			clusterWide, err := cmd.Flags().GetBool("cluster")
+	RunE: func(cmd *cobra.Command, args []string) error {
+		level.SetZeroLogLevel()
+		namespace, err := cmd.Flags().GetString("namespace")
+		if err != nil {
+			return err
+		}
+		kubewardenNamespace, err := cmd.Flags().GetString("kubewarden-namespace")
+		if err != nil {
+			return err
+		}
+		clusterWide, err := cmd.Flags().GetBool("cluster")
+		if err != nil {
+			return err
+		}
+		policyServerURL, err := cmd.Flags().GetString("policy-server-url")
+		if err != nil {
+			return err
+		}
+		caCertFile, err := cmd.Flags().GetString("extra-ca")
+		if err != nil {
+			return err
+		}
+		storeType, err := cmd.Flags().GetString("store")
+		if err != nil {
+			return err
+		}
+
+		policyReportStore, err := getPolicyReportStore(storeType)
```

Comment:
We should be creating the store in scanner.go as we were doing before, and not in `cmd/root.go`. Ideally, `cmd/root.go` should be only used for reading the cmd flags, and calling an audit-scanner lnternal library with all the logic. Right now `cmd/root.go` already does more than it should (create policiesFetcher, resourcesFetcher, etc) but that should change in the future.

Refactoring into an internal lib would simplify testing and mocks, yet it is out of the scope for this PR. But that doesn't mean we shouldn't keep piling up things in `cmd/root.go`.

Reply:
I completely agree with you, in this case, I went to the easy path and decided to add another dependency to `cmd/root.go` when shouldn't. I've refactored this bit with your suggestions. there's still `outputScan` there that I'm happy to move it elsewhere after https://github.com/kubewarden/audit-scanner/pull/150#discussion_r1417210469